src/test/java/api/subscribers/CloseDeclaredLostLoanWhenLostItemFeesAreClosedApiTests.java

```diff
@@ -0,0 +1,141 @@
+package api.subscribers;
+
+import static api.support.matchers.ItemMatchers.isAvailable;
+import static api.support.matchers.ItemMatchers.isCheckedOut;
+import static api.support.matchers.ItemMatchers.isDeclaredLost;
+import static api.support.matchers.ItemMatchers.isLostAndPaid;
+import static api.support.matchers.JsonObjectMatcher.hasJsonPath;
+import static api.support.matchers.LoanMatchers.isClosed;
+import static api.support.matchers.LoanMatchers.isOpen;
+import static api.support.matchers.ValidationErrorMatchers.hasErrorWith;
+import static api.support.matchers.ValidationErrorMatchers.hasMessage;
+import static api.support.matchers.ValidationErrorMatchers.hasNullParameter;
+import static org.hamcrest.CoreMatchers.allOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.UUID;
+
+import org.folio.circulation.support.http.client.IndividualResource;
+import org.folio.circulation.support.http.client.Response;
+import org.junit.Before;
+import org.junit.Test;
+
+import api.support.APITests;
+import api.support.builders.DeclareItemLostRequestBuilder;
+
+public class CloseDeclaredLostLoanWhenLostItemFeesAreClosedApiTests extends APITests {
+  private IndividualResource loan;
+  private IndividualResource item;
+
+  @Before
+  public void createLoanAndDeclareItemLost() {
+    useLostItemPolicy(lostItemFeePoliciesFixture.chargeFee().getId());
+
+    item = itemsFixture.basedUponSmallAngryPlanet();
+    loan = checkOutFixture.checkOutByBarcode(item, usersFixture.jessica());
+
+    declareLostFixtures.declareItemLost(new DeclareItemLostRequestBuilder()
+      .forLoanId(loan.getId()));
+  }
+
+  @Test
+  public void shouldCloseLoanWhenAllFeesClosed() {
+    feeFineAccountFixture.payLostItemFee(loan.getId());
+    feeFineAccountFixture.payLostItemProcessingFee(loan.getId());
+
+    eventSubscribersFixture.publishLoanRelatedFeeFineClosedEvent(loan.getId());
+
+    assertThat(loansFixture.getLoanById(loan.getId()).getJson(), isClosed());
+    assertThat(itemsClient.getById(item.getId()).getJson(), isLostAndPaid());
+  }
+
+  @Test
+  public void shouldNotCloseLoanWhenProcessingFeeIsNotClosed() {
+    feeFineAccountFixture.payLostItemFee(loan.getId());
+
+    eventSubscribersFixture.publishLoanRelatedFeeFineClosedEvent(loan.getId());
+
+    assertThat(loansFixture.getLoanById(loan.getId()).getJson(), isOpen());
+    assertThat(itemsClient.getById(item.getId()).getJson(), isDeclaredLost());
+  }
+
+  @Test
+  public void shouldNotCloseLoanIfItemFeeIsNotClosed() {
```

Comment:
Is this a set cost item fee?

Reply:
yes

Reply:
Maybe that could be stated in the name?

```suggestion
  public void shouldNotCloseLoanIfSetCostFeeIsNotClosed() {
```